lib/libc/minimal/source/thrd/thrd.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <threads.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/posix/pthread.h>
+
+struct thrd_trampoline_arg {
+	thrd_start_t func;
+	void *arg;
+};
+
+int thrd_create(thrd_t *thr, thrd_start_t func, void *arg)
+{
+	typedef void *(*pthread_func_t)(void *arg);
+
+	pthread_func_t pfunc = (pthread_func_t)func;
+
+	switch (pthread_create(thr, NULL, pfunc, arg)) {
+	case 0:
+		return thrd_success;
+	case EAGAIN:
+		return thrd_nomem;
+	default:
+		return thrd_error;
+	}
+}
+
+int thrd_equal(thrd_t lhs, thrd_t rhs)
+{
+	return pthread_equal(lhs, rhs);
+}
+
+thrd_t thrd_current(void)
+{
+	return pthread_self();
+}
+
+int thrd_sleep(const struct timespec *duration, struct timespec *remaining)
+{
+	return nanosleep(duration, remaining);
+}
+
+void thrd_yield(void)
+{
+	k_yield();
```

Comment:
For consistent layering, this should be replaced with `pthread_yield()`, but it's effectively the same thing.

Reply:
how about [`sched_yield`](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/posix/sched.h#L40)?

Reply:
I think cfriedt wanted to ensure that the whole c11 thread API was implemented in terms of standard pthreads APIs; poking through to `k_yield` exposes a gap in that layer.

Reply:
I did add https://github.com/zephyrproject-rtos/zephyr/issues/60757 but then closed it shortly after ycsin told me it was deprecated, and it's no longer on the opengroup site.

For reference
https://man7.org/linux/man-pages/man3/pthread_yield.3.html

I guess I don't have anything against adding `pthread_yield()` even if it is deprecated, but technically `sched_yield()` would work too.

Reply:
Fixed!